### ISSUE TITLE
MPCM catalog layer access specification

### ISSUE THREAD

**Issue by NicoledeGreef** [bcgov/smk-legacy-ui#81]
Upon review of 1.1.0 in smk-demo, I see that I can currently reach Diagnostic Facilities (Internal Access) layers which are sensitive and add them to a new SMK map.  
As IDIR\NDEGREEF I do have access to those layers but public consumers of the map should not.
@cpspicer - do we need the SMK site authoring UI to reach only public layers?

**Comment by cpspicer** [comment]
Correct - public layers only at this time.  The MPCM endpoint won't allow non-authenticated users to access the content, but we don't want app authors with higher privs to be able to add them into their apps.  SMK is public only content for now.

**Comment by dhlevi** [comment]
A filter has been added to remove and layers that are secure, however note that this is a manual filter in the service side as the MPCM rest service doesn't return workspace info unless you call the layers directly. We can prevent adding secure layers, but to hide them in the list before making the full request is less then optimal.

**Comment by NicoledeGreef** [comment]
So what is the nature of the filter that was applied?  Is it searching for the word "internal"? 

Some secured layers are being omitted from the list but some are not. 

e.g. I can attempt to add "Electoral Districts - 2017" and the SMK authoring tool will prevent me from adding it as it is MPCM_EBC_DBC not MPCM_ALL_PUB.

**Comment by dhlevi** [comment]
Thats correct. Initially the list will strip out anything that contains "internal", which of course is not going to catch everything. When you attempt to add a layer the service can then check the workspace, and if it isn't public, you wont be abe to add it.

This will not prevent anyone from adding a secure layer manually after exporting.
